x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/conditional/Case.java

```diff
@@ -161,14 +161,24 @@ public ScriptTemplate asScript() {
         }
         templates.add(asScript(elseResult));
 
-        StringJoiner template = new StringJoiner(",", "{sql}.caseFunction([", "])");
+        // Use painless if-else expressions to prevent evaluation of return expression
+        // if the condition is guarded by evaluates to false (e.g. division by 0)
+        StringBuilder sb = new StringBuilder();
         ParamsBuilder params = paramsBuilder();
-
-        for (ScriptTemplate scriptTemplate : templates) {
-            template.add(scriptTemplate.template());
+        for (int i = 0; i < templates.size(); i++) {
+            ScriptTemplate scriptTemplate = templates.get(i);
+            if (i < templates.size() - 1) {
```

Comment:
Why is nullSafeFilter added at this level  - we already add it (or should do so) on the top-level (`ScriptQuery`), why does this has to happen inside `Case`?

Reply:
Because the painless `? :` operator complains when receives a `Boolean` object instead of primitive `boolean`.

Reply:
I see - the nullSafe then is a good wrapper. Please add a comment since this Painless behavior is not obvious looking at the assembly.